endpoints.go

```
@@ -40,6 +42,34 @@ func endpoints(gcfg globalConfig) ([]string, error) {
 	return endpointsFromCluster(gcfg)
 }
 
+func IsLocalEp(ep string) bool {
+	if strings.HasPrefix(ep, "unix:") || strings.HasPrefix(ep, "unixs:") {
+		return true
+	}
+
+	if strings.Contains(ep, "://") {
+		url, err := url.Parse(ep)
+		if err != nil {
+			return false
+		}
+		if url.Scheme == "http" || url.Scheme == "https" {
```

If the url.Scheme is neither http nor https, we should return an error.